doc/paper/paper.tex

```diff
@@ -0,0 +1,1012 @@
+    files, and manifest files contain a list of data files in Avro format.
+    Iceberg has built-in support for Parquet, Avro, and ORC format data files. As
+    with dp3 and btrdb, it seems logical in deployment scenarios to cache the
+    upper nodes of the tree.
+
+    The primary intent of Iceberg is to enable multiple database engines to
+    operate against the same data safely at the same time, but as a useful
+    side-effect of having data in Iceberg format you gain the ability to use an
+    off-the-shelf database engine for free, and the potential for pluggable
+    integrations in the future if more databases add support for external Iceberg
+    tables, as Snowflake has recently done. In addition, dp3 may at some point
+    want to have multiple engines operating at the same time, for instance if we
+    were able to integrate with Spark SQL. These possibilities make Iceberg a
+    worthwhile direction to explore as backing storage for dp3. In comparing the
+    two, Iceberg stands out for its far more developed SQL and transactional
+    capabilities, while dp3 is distinguished in supporting message storage and
+    playback in the user's native choice of recording format, without on-the-fly
+    transcoding during read. dp3 also has a more granular time index due to
+    having a deeper tree.
+
+    The query language and goals of dp3 bear similarity to query languages and
+    systems previously developed in the areas of \q{stream databases} and
+    \q{sequence databases}, such as CQL \cite{cql}, the GSQL language for the
+    Gigascope database \cite{gigascope}, Aurora \cite{aurora}, and SEQUIN for
+    the SEQ system. These languages are broadly SQL-like with a focus on
+    support for merging and as-of join operations, although those go by
+    different names.
+
+    CQL has a focus on continuous queries over unbounded data streams. CQL
+    supports a \q{sliding window join} with the syntax 
+
+    \begin{lstlisting}
+    Select Istream(*)
+    From S1 [Range 5 seconds], S2 [Range 10 seconds]
+    Where S1.A = S2.A
+    \end{lstlisting}
+
+    in this example, a tuple from S1 will be sent to the output if it joins with
+    a tuple in S2 from within a 10-second sliding window, and a tuple from S2 will
+    be sent if it joins with a 5-second sliding window from S1.
+
+    dp3 does not support this kind of bidirectional windowing today, but this
+    query is pretty similar to what dp3 would express as
+
+    \begin{lstlisting}
+    from my-device
+    S1 precedes S2 by less than 10 seconds
+    where S1.field = "foo"
+    and S2.field = "bar"
+    \end{lstlisting}
+
+    The concept of joining columns between tables in dp3 is not presumed to make
+    sense today, because the fields of different tables are typically
+    incomparable. However, support for this may become important as we develop
+    more focus on subqueries.
+
+    Continuous queries are also not supported in dp3 today, however support
```

Comment:
--> "; however, support would not require much work."
Should we mention this in our feature development plans if not mentioned already?